dockerfile

```diff
@@ -0,0 +1,17 @@
+FROM python:3.11
+
+
+WORKDIR /usr/src/app
+
+
+COPY pyproject.toml ./
+
+# Install poetry for package management
+RUN pip install --no-cache-dir poetry
+RUN poetry install
+
+
+COPY . .
```

Comment:
**suggesion(non-blocking):** Do all files need to be copied into container? You can use `.dockerignore` to exclude some. See: https://docs.docker.com/build/building/context/#dockerignore-files